src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/Base64Transforms.cs

```diff
@@ -0,0 +1,294 @@
+        {
+            _whitespaces = whitespaces;
+            _inputIndex = 0;
+        }
+
+        // Converting from Base64 generates 3 bytes output from each 4 bytes input block
+        public int InputBlockSize
+        {
+            get { return (1); }
+        }
+
+        public int OutputBlockSize
+        {
+            get { return (3); }
+        }
+
+        public bool CanTransformMultipleBlocks
+        {
+            get { return (false); }
+        }
+
+        public virtual bool CanReuseTransform
+        {
+            get { return (true); }
+        }
+
+        public int TransformBlock(byte[] inputBuffer, int inputOffset, int inputCount, byte[] outputBuffer, int outputOffset)
+        {
+            // Do some validation
+            if (inputBuffer == null) throw new ArgumentNullException(nameof(inputBuffer));
+            if (inputOffset < 0) throw new ArgumentOutOfRangeException(nameof(inputOffset), SR.ArgumentOutOfRange_NeedNonNegNum);
+            if (inputCount < 0 || (inputCount > inputBuffer.Length)) throw new ArgumentException(SR.Argument_InvalidValue);
+            if ((inputBuffer.Length - inputCount) < inputOffset) throw new ArgumentException(SR.Argument_InvalidOffLen);
+
+            if (_inputBuffer == null)
+                throw new ObjectDisposedException(null, SR.ObjectDisposed_Generic);
+
+            byte[] temp = new byte[inputCount];
+            char[] tempChar;
+            int effectiveCount;
+
+            if (_whitespaces == FromBase64TransformMode.IgnoreWhiteSpaces)
+            {
+                temp = DiscardWhiteSpaces(inputBuffer, inputOffset, inputCount);
+                effectiveCount = temp.Length;
+            }
+            else
+            {
+                Buffer.BlockCopy(inputBuffer, inputOffset, temp, 0, inputCount);
+                effectiveCount = inputCount;
+            }
+
+            if (effectiveCount + _inputIndex < 4)
+            {
+                Buffer.BlockCopy(temp, 0, _inputBuffer, _inputIndex, effectiveCount);
+                _inputIndex += effectiveCount;
+                return 0;
+            }
+
+            // Get the number of 4 bytes blocks to transform
+            int numBlocks = (effectiveCount + _inputIndex) / 4;
+            byte[] transformBuffer = new byte[_inputIndex + effectiveCount];
+            Buffer.BlockCopy(_inputBuffer, 0, transformBuffer, 0, _inputIndex);
+            Buffer.BlockCopy(temp, 0, transformBuffer, _inputIndex, effectiveCount);
+            _inputIndex = (effectiveCount + _inputIndex) % 4;
+            Buffer.BlockCopy(temp, effectiveCount - _inputIndex, _inputBuffer, 0, _inputIndex);
+
+            tempChar = Encoding.ASCII.GetChars(transformBuffer, 0, 4 * numBlocks);
+
+            byte[] tempBytes = Convert.FromBase64CharArray(tempChar, 0, 4 * numBlocks);
+            Buffer.BlockCopy(tempBytes, 0, outputBuffer, outputOffset, tempBytes.Length);
```

Comment:
In neither to To nor From case do we check that outputBuffer isn't null, nor that it's long enough.  Since that's matching Desktop behaviors please add tests to codify our behaviors.